### ISSUE TITLE
can‘t download, login failed

### ISSUE THREAD

**Issue by Choogle-Ma** [ClaudiuGeorgiu/PlaystoreDownloader#34]
I run the script and get the failed results. 
I download apps successfully in real phone using the same accounts.

What's wrong?

`ec2-user@ip-172-31-43-179:~/project/PlaystoreDownloader-master> python3 download.py "com.android.vending"
01/04/2020 10:16:14> [WARNING][playstore.util][wrapped()] Login failed, please check your credentials (retrying in 1s)
01/04/2020 10:16:15> [WARNING][playstore.util][wrapped()] Login failed, please check your credentials (retrying in 2s)
01/04/2020 10:16:17> [WARNING][playstore.util][wrapped()] Login failed, please check your credentials (retrying in 3s)
01/04/2020 10:16:20> [WARNING][playstore.util][wrapped()] Login failed, please check your credentials (retrying in 5s)
`

**Comment by ClaudiuGeorgiu** [comment]
Hi, in the logs I see `ec2-user@ip-172-31-43-179` so I guess you're using a remote server. Have you tried downloading the apps on your local network (the same used by your phone, if that's possible)? Did you get any notification from the Google account saying saying that there's some suspicious activity?
When I get the login error, usually it means that I'm trying to use the script on a remote server somewhere far from my real location and I also get a security notification by Google (that can be temporary solved by visiting [this link](https://accounts.google.com/DisplayUnlockCaptcha) while being logged with the account used to download the app).

**Comment by Choogle-Ma** [comment]
Yes I got the suspicious activity notification. but after I visit the unlocked page to verify the account. I still can not download it.

**Comment by HuErr** [comment]
Account opened with two-step verification

**Comment by marekceglowski** [comment]
@ClaudiuGeorgiu 
Hi, is the app currently working? I am getting login issues as well. I have 2-factor authentication disabled, I am using the proper Google Service Framework (GSF) value as my ANDROID_ID, the username and password are correct, and I enabled the https://accounts.google.com/DisplayUnlockCaptcha and https://myaccount.google.com/lesssecureapps as suggested. It still produces the error:

```
python download.py "com.td.myadvantage"
26/05/2020 23:00:28> [WARNING][playstore.util][wrapped()] Login failed, please check your credentials (retrying in 1s)
26/05/2020 23:00:29> [WARNING][playstore.util][wrapped()] Login failed, please check your credentials (retrying in 2s)
26/05/2020 23:00:32> [WARNING][playstore.util][wrapped()] Login failed, please check your credentials (retrying in 3s)
26/05/2020 23:00:35> [WARNING][playstore.util][wrapped()] Login failed, please check your credentials (retrying in 5s)
26/05/2020 23:00:40> [ERROR][playstore.util][wrapped()] Login failed, please check your credentials (no more retries)
26/05/2020 23:00:40> [CRITICAL][__main__][main()] Error during the download: Login failed, please check your credentials
```
Any suggestions? Thanks.

**Comment by ClaudiuGeorgiu** [comment]
Hi @marekceglowski, I just tried to download some apps and I can confirm that (in my case) it's working. Are you able to download the app without issues when using directly your device? Are you trying to use the tool on the same network as your device (e.g., your home wifi)? Did you get any security notification from the Google account? You could try adding `print(response.text)` after this line:
https://github.com/ClaudiuGeorgiu/PlaystoreDownloader/blob/574fe5157f8ea5264df31e9f2783425b05cd8f95/playstore/playstore.py#L100
This should give you more information about the actual error.

**Comment by leon1995** [comment]
Hi. I have the same issues when using the docker image. I followed alle the steps as mentioned and tried to download an apk from the same network. Still get the login error.

When I try to use the python script I get a `No module named Crypto` even tho I installed all packages in the requirenments from the requirenments file and tried to install it with `pip3 install crypto`

**Comment by thecoderok** [comment]
I have same issue. I added `print(response.text)` to playstore.py, here is what I have:
```
24/06/2020 13:29:32> [ERROR][playstore.util][wrapped()] Login failed, please check your credentials: {'error': 'BadAuthentication'} (no more retries)
24/06/2020 13:29:32> [CRITICAL][__main__][main()] Error during the download: Login failed, please check your credentials: {'error': 'BadAuthentication'}
(venv) vmganzha-mbp:PlaystoreDownloader vmganzha$ python3 download.py "com.spotify.music"
Response: 'Error=BadAuthentication'
24/06/2020 13:29:54> [WARNING][playstore.util][wrapped()] Login failed, please check your credentials: {'error': 'BadAuthentication'} (retrying in 1s)
Response: 'Error=BadAuthentication'
24/06/2020 13:29:55> [WARNING][playstore.util][wrapped()] Login failed, please check your credentials: {'error': 'BadAuthentication'} (retrying in 2s)
Response: 'Error=BadAuthentication'
24/06/2020 13:29:58> [WARNING][playstore.util][wrapped()] Login failed, please check your credentials: {'error': 'BadAuthentication'} (retrying in 3s)
```

**Comment by Hanzelka** [comment]
Hi, I am facing the same issue. 

docker run     -u $(id -u):$(id -g)     -v "${PWD}/credentials.json":"/app/credentials.json"     -v "${PWD}/output/":"/app/Downloads/"     --rm -it downloader "com.spotify.music"

07/10/2020 08:12:43> [ERROR][playstore.playstore.Playstore][download()] Error during the download: [Errno 13] Permission denied: 'Downloads/Spotify_ Listen to new music_ podcasts_ and songs by Spotify Ltd. - com.spotify.music.apk'
Traceback (most recent call last):
  File "/app/playstore/playstore.py", line 684, in download
    list(
  File "/app/playstore/playstore.py", line 371, in _download_with_progress
    yield from self._download_single_file(
  File "/app/playstore/playstore.py", line 194, in _download_single_file
    with open(destination_file, "wb") as f:
PermissionError: [Errno 13] Permission denied: 'Downloads/Spotify_ Listen to new music_ podcasts_ and songs by Spotify Ltd. - com.spotify.music.apk'

07/10/2020 08:12:43> [CRITICAL][__main__][main()] Error when downloading 'com.spotify.music'

**Comment by ClaudiuGeorgiu** [comment]
Hi @Hanzelka, you don't have a login error but a file permission error, maybe because you don't have an `output/` folder in the current directory. Try the following command:
```
docker run -v "${PWD}/credentials.json":"/app/credentials.json" --rm -it downloader "com.spotify.music"
```
If it works, then the problem is caused by a bad docker volume configuration.

**Comment by Hanzelka** [comment]
@ClaudiuGeorgiu 
Thanks for quick response. You are right, I had a wrong permissions on output folder. I am able to download apps now. 

Thank you

**Comment by ClaudiuGeorgiu** [comment]
This should be solved by bb5f59ab3be52f53ea0b15bf2b5ebf8de7b0925a.
